kms/main.tf

```diff
@@ -0,0 +1,140 @@
+data "aws_caller_identity" "current" {}
+
+resource "aws_kms_key" "encryption" {
+  description         = var.key_description
+  enable_key_rotation = true
+  policy              = var.key_policy == "" ? data.aws_iam_policy_document.key_policy.json : var.key_policy
+  tags = merge(
+    var.tags,
+    tomap(
+      { "Name" = var.key_name }
+    )
+  )
+}
+
+resource "aws_kms_alias" "encryption" {
+  name          = "alias/${var.key_name}"
+  target_key_id = aws_kms_key.encryption.key_id
+}
+
+module "kms_admin_role" {
+  source             = "../iam_role"
+  assume_role_policy = templatefile("${path.module}/templates/deny_all.json.tpl", {})
+  name               = "${var.key_name}-admin"
+  policy_attachments = {
+    kms_power_user = "arn:aws:iam::aws:policy/AWSKeyManagementServicePowerUser"
+  }
+  tags = merge(
+    var.tags,
+    tomap(
+      { "Name" = "${var.key_name}-admin" }
+    )
+  )
+}
+
+data "aws_iam_policy_document" "key_policy" {
```

Comment:
Is it worth putting `Sid` properties in the statements to describe what they're doing for users of this module?